### ISSUE TITLE
MSRV attribute is not scoped

### ISSUE THREAD

**Issue by camsteffen** [rust-lang/rust-clippy#6920]
Example:

```rust
#![feature(custom_inner_attributes)]

mod my_mod {
    #![clippy::msrv = "1.0.0"]
}

fn hi() {
    // the `manual_strip` lint is suppressed by the attribute in `my_mod`
    let s = "hello, world!";
    if s.starts_with("hello, ") {
        assert_eq!(s["hello, ".len()..].to_uppercase(), "WORLD!");
    }
}
```

If we don't want to support this sort of usage, maybe we can just warn or error when the attribute is found in a non-crate.

**Comment by hellow554** [comment]
There's this testfile https://github.com/rust-lang/rust-clippy/blob/master/tests/ui/min_rust_version_attr.rs and it seems that scoping is supported in some way.

It seems, that the inner attribute starts from that line on in the code, because if I move your `hi` function above `my_mod` the lint is triggered correctly.
